### PR TITLE
Fixed-new-email-api

### PR DESCRIPTION
**Pull request by saranyadevi-bc**

### CODE REVIEW

File: Ccs.Ppg.NotificationService.API/CustomOptions/VaultConfiguration.cs [hunk 1]
```diff
@@ -56,7 +56,14 @@ public async Task GetSecrets()
         Data.Add("Message:TemplateId", messageSettingsVault.TemplateId);
       }
 
-      if (_secrets.Data.ContainsKey("ConnectionStrings"))
+			if (_secrets.Data.ContainsKey("Email"))
+			{
+				var messageSettingsVault = JsonConvert.DeserializeObject<MessageSettingsVault>(_secrets.Data["Email"].ToString());
+				Data.Add("Email:ApiKey", messageSettingsVault.ApiKey);
+				Data.Add("Email:TemplateId", messageSettingsVault.TemplateId);
```
Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService.API/CustomOptions/VaultConfiguration.cs:63]: Remove this line

File: Ccs.Ppg.NotificationService.API/appsecrets.json [hunk 2]
```diff
@@ -14,6 +14,10 @@
   },
   "apis": {
     "OrganisationUrl": ""
+  },
+  "Email": {
+    "ApiKey": "",
+    "TemplateId": ""
```
Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService.API/appsecrets.json:20]: Remove this line

File: Ccs.Ppg.NotificationService.API/Controllers/NotificationController.cs [hunk 3]
```diff
@@ -45,5 +47,13 @@ public async Task<bool> Post(MessageRequestModel message)
         {
             return await _messageProviderService.SendMessage(message);
         }
-    }
-}
+
```
Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService.API/Controllers/NotificationController.cs:50]: add description for this API, take reference of sms API

File: Ccs.Ppg.NotificationService/Services/EmailProviderService.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+		public EmailProviderService(IConfiguration configuration, IHttpClientFactory httpClientFactory)
+		{
+			_configuration = configuration;
+			_httpClientFactory = httpClientFactory;
+		}
+		public async Task SendEmailAsync(EmailInfo emailInfo)
+		{
+			var apiKey = _configuration["Email:ApiKey"];
+			var templateId = !string.IsNullOrEmpty(emailInfo.TemplateId) ? emailInfo.TemplateId : _configuration["emailInfo:TemplateId"];
```
Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService/Services/EmailProviderService.cs:23]: we will not have TemplateId in config..

File: Ccs.Ppg.NotificationService/Services/EmailProviderService.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	{
+		private readonly IConfiguration _configuration;
+		private readonly IHttpClientFactory _httpClientFactory;
+
+		public EmailProviderService(IConfiguration configuration, IHttpClientFactory httpClientFactory)
+		{
+			_configuration = configuration;
+			_httpClientFactory = httpClientFactory;
+		}
+		public async Task SendEmailAsync(EmailInfo emailInfo)
```
Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService/Services/EmailProviderService.cs:20]: add try catch in this method, refer sms method.

Comment (brijeshpatel-bc) [inline Ccs.Ppg.NotificationService/Services/EmailProviderService.cs:20]: add required validation for email fields like To, TemplateId

File: Ccs.Ppg.NotificationService.API/CustomOptions/VaultConfiguration.cs (hunk 1 above)
Comment (saranyadevi-bc) [reply Ccs.Ppg.NotificationService.API/CustomOptions/VaultConfiguration.cs:63]: Done

File: Ccs.Ppg.NotificationService.API/Controllers/NotificationController.cs (hunk 3 above)
Comment (saranyadevi-bc) [reply Ccs.Ppg.NotificationService.API/Controllers/NotificationController.cs:50]: Done

File: Ccs.Ppg.NotificationService/Services/EmailProviderService.cs (hunk 5 above)
Comment (saranyadevi-bc) [reply Ccs.Ppg.NotificationService/Services/EmailProviderService.cs:20]: Done

File: Ccs.Ppg.NotificationService/Services/EmailProviderService.cs (hunk 4 above)
Comment (saranyadevi-bc) [reply Ccs.Ppg.NotificationService/Services/EmailProviderService.cs]: Okay

**Review by brijeshpatel-bc — APPROVED** [review]

**Pull request merged**